### PR TITLE
Fix HTTP/1 TestServer flakiness

### PR DESCRIPTION
**Pull request by kgiusti**
Use context managers for both the TestServer and its internal HTTPServer instance.
Fix ISSUE #70

### CODE REVIEW

File: tests/http1_tests.py [hunk 1]
```diff
@@ -325,9 +331,12 @@ def _send_shutdown_request(self):
...
 
-    def _force_disconnect_clients(self):
-        """Shutdowns the client socket in the http.server"""
-        self._server.force_disconnect_clients()
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        if self._thread.is_alive():
+            self.wait()
```
Comment (jiridanek) [inline tests/http1_tests.py:339]: 👍🏽

File: tests/http1_tests.py [hunk 2]
```diff
@@ -264,46 +264,52 @@ def new_server(cls, server_port, client_port, tests, handler_cls=None):
             raise RuntimeError(f"TestServer failed to start due to port {server_port} already in use issue") from exc
 
     def __init__(self, server_port, client_port, tests, handler_cls=None):
```
Comment (ganeshmurthy) [inline tests/http1_tests.py:266]: This is minor but can we please move the `__init__` function as the first function in this class so I can immediately see it?

File: tests/http1_tests.py [hunk 3]
```diff
@@ -264,46 +264,52 @@ def new_server(cls, server_port, client_port, tests, handler_cls=None):
...
-                                    tests)
-        self._server.allow_reuse_address = True
-        self._server.timeout = TIMEOUT
-        self._thread = Thread(target=self._run)
+        self._server_error = None
+        self._is_ready = Event()
+        self._thread = Thread(target=self._run, args=(tests,
+                                                      handler_cls or RequestHandler))
         self._thread.start()
+        self._is_ready.wait(TIMEOUT)
```
Comment (ganeshmurthy) [inline tests/http1_tests.py:276]: Can you please add a comment in here explaining why a `wait()` is called here and the `set()` is called from inside the thread ?

**Comment by kgiusti** [comment]
@jiridanek Hey Jiri: it looks like the router log files are not included in the artifacts for the test that failed https://github.com/skupperproject/skupper-router/runs/6102760939?check_suite_focus=true#step:26:2262

Did something go wrong with the upload?  I don't see the "tests/..." directory in the uploaded tar...

**Comment by jiridanek** [comment]
> Artifact testLogs_fedora_RelWithDebInfo_tsan_main_1 has been successfully uploaded!

That looks ok...

But above that

```
Run python3 /home/runner/work/skupper-router/skupper-router/skupper-router/scripts/gha_purge_successful_test_logs.py --build-dir=/home/runner/work/skupper-router/skupper-router/skupper-router/build --no-dry-run
INFO:root:deleted /home/runner/work/skupper-router/skupper-router/skupper-router/build/tests/system_test.dir/tests
```

That is... weird

**Review by ganeshmurthy — APPROVED** [review]

**Comment by jiridanek** [comment]
> That is... weird

https://github.com/skupperproject/skupper-router/pull/361

**Pull request merged**